src/stavisky_lab_to_nwb/stavisky/staviskysortinginterface.py

```diff
@@ -1,23 +1,136 @@
+        This function should retrieve the data on-demand by re-initializing the IO.
 
-    def add_to_nwbfile(self, nwbfile: NWBFile, metadata: dict):
-        # All the custom code to write to PyNWB
+        Returns
+        -------
+        timestamps: numpy.ndarray
+            The timestamps for the data stream.
+        """
+        return self.sorting_extractor.get_ids_and_timestamps(
+            stream_name=self.source_data.get("stream_name"),
+            frames_per_entry=self.source_data.get("frames_per_entry", 1),
+            sampling_frequency=self.source_data.get("sampling_frequency", None),
+            timestamp_source=self.source_data.get("timestamp_source", None),
+            **self.source_data.get("timestamp_kwargs", dict()),
+        )[1]
 
-        return nwbfile
+    def get_timestamps(self) -> np.ndarray:
+        """
+        Retrieve the timestamps for the data in this interface.
+
+        Returns
+        -------
+        timestamps: numpy.ndarray
+            The timestamps for the data stream.
+        """
+        return self.sorting_extractor.get_times()
+
+    def set_aligned_timestamps(self, aligned_timestamps: np.ndarray) -> None:
+        """
+        Replace all timestamps for this interface with those aligned to the common session start time.
+
+        Must be in units seconds relative to the common 'session_start_time'.
+
+        Parameters
+        ----------
+        aligned_timestamps : numpy.ndarray
+            The synchronized timestamps for data in this interface.
+        """
+        # Removed requirement of having recording
+        if self._number_of_segments == 1:
+            self.sorting_extractor.set_times(times=aligned_timestamps)
+        else:
+            assert isinstance(
+                aligned_timestamps, list
+            ), "Recording has multiple segment! Please pass a list of timestamps to align each segment."
+            assert (
+                len(aligned_timestamps) == self._number_of_segments
+            ), f"The number of timestamp vectors ({len(aligned_timestamps)}) does not match the number of segments ({self._number_of_segments})!"
+
+            for segment_index in range(self._number_of_segments):
+                self.sorting_extractor.set_times(times=aligned_timestamps[segment_index], segment_index=segment_index)
+
+    def set_aligned_starting_time(self, aligned_starting_time: float) -> None:
+        """
+        Align the starting time for this interface relative to the common session start time.
+
+        Must be in units seconds relative to the common 'session_start_time'.
+
+        Parameters
+        ----------
+        aligned_starting_time : float
+            The starting time for all temporal data in this interface.
+        """
+        self.set_aligned_timestamps(aligned_timestamps=self.get_timestamps() + aligned_starting_time)
+
+    def align_by_interpolation(self, unaligned_timestamps: np.ndarray, aligned_timestamps: np.ndarray) -> None:
```

Comment:
There shouldn't be any need to define `set_aligned_starting_time` or `align_by_interpolation`, since those are inherently defined in terms of the other methods at the [base level in NeuroConv](https://github.com/catalystneuro/neuroconv/blob/main/src/neuroconv/basetemporalalignmentinterface.py)

Reply:
ah yeah whoops, for some reason I thought `BaseSortingExtractor` didn't inherit from `BaseTemporalAlignmentInterface`, will remove those

Reply:
actually, I can remove `align_by_interpolation`, but `BaseSortingExtractor` overrides `set_aligned_starting_time` so I'd need to keep my changes for that function in. I could explicitly call `BaseTemporalAlignmentInterface.set_aligned_starting_time()`, but it wouldn't really save any space so I just kept it as is for now